### PR TITLE
[op-conductor] e2e tests part 1 - test setup

### PR DESCRIPTION
**Pull request by 0x00101010**
**Description**

1. Add e2e tests to test conductor functionalities along with 3 sequencer setup HA transitions, failure recoveries.
2. Fix `TestPostUnsafePayload` flakiness
3. Improve `op-service/rpc.Server` to allow passing in 0 port and get back the real assigned port in tests

**Tests**

This PR mainly sets up a working HA cluster for further tests, more tests will be added in follow up PRs which includes:
1. conductor rpc tests
4. failover scenario tests
5. disaster recovery tests

**Metadata**

- https://github.com/ethereum-optimism/protocol-quest/issues/85

### CODE REVIEW

File: op-e2e/sequencer_failover_test.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// In this test, we test that we can successfully setup a working cluster.
+func TestSequencerFailver_SetupCluster(t *testing.T) {
+	sys, conductors := setupSequencerFailoverTest(t)
+	defer sys.Close()
+
+	require.Equal(t, 3, len(conductors), "Expected 3 conductors")
+	for _, con := range conductors {
+		require.NotNil(t, con, "Expected conductor to be non-nil")
+	}
+}
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply op-e2e/sequencer_failover_test.go]: Fixed

File: op-e2e/sequencer_failover_setup.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+	rawClient, err := rpc.DialContext(ctx, service.HTTPEndpoint())
+	require.NoError(t, err)
+	client := conrpc.NewAPIClient(rawClient)
+
+	return &conductor{
+		service: service,
+		client:  client,
+	}
+}
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply op-e2e/sequencer_failover_setup.go]: This is not necessary, current approach is cleaner, we can separate it into a separate function if setup requires more steps

File: op-conductor/conductor/service.go [hunk 3]
```diff
@@ -329,6 +329,13 @@ func (oc *OpConductor) Paused() bool {
...
+func (oc *OpConductor) HTTPEndpoint() string {
+	if oc.rpcServer == nil {
+		return ""
+	}
+	return fmt.Sprintf("http://%s", oc.rpcServer.Endpoint())
+}
+
 // Leader returns true if OpConductor is the leader.
 func (oc *OpConductor) Leader(_ context.Context) bool {
 	return oc.cons.Leader()
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply op-conductor/conductor/service.go:341]: This is not valid suggestion, we queue action without knowing what will be attempted.

File: op-e2e/sequencer_failover_setup.go [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+			if err != nil {
+				return err
+			}
+			if isLeader == leader {
+				return nil
+			}
+			time.Sleep(500 * time.Millisecond)
+		}
+	}
+}
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply op-e2e/sequencer_failover_setup.go:292]: No backoff needed

File: op-service/rpc/server.go [hunk 5]
```diff
@@ -180,14 +181,22 @@ func (b *Server) Start() error {
 	handler = oplog.NewLoggingMiddleware(b.log, handler)
 	b.httpServer.Handler = handler
 
+	listener, err := net.Listen("tcp", b.endpoint)
```
Comment (0x00101010) [inline op-service/rpc/server.go:184]: Adding listener here allows us to pass in `0` port in test and get back the actual assigned port from `Endpoint()`

**Review by zhwrd — APPROVED** [review]

**Review by trianglesphere — APPROVED** [review]

**Pull request merged**